### DISCUSSION TITLE
How to Build a Custom Heterogeneous Graph Dataset for Railway Scenarios

### DISCUSSION THREAD

**Discussion by baidingyuan** [pyg-team/pytorch_geometric#9507] [Q&A]
### Description:
Hello,

I am looking to create a custom dataset for railway scenarios, specifically a heterogeneous graph dataset. The goal is to describe the various equipment in railway scene images and their relationships. For example, relationships like "train on the track" or "pedestrian beside the track". However, I am unsure how to start building such a dataset and would appreciate any guidance or advice.

### Steps to Reproduce:

Identify the key elements in railway scene images (e.g., trains, tracks, pedestrians, signals, etc.).
Define the relationships between these elements (e.g., "train on track", "pedestrian beside track").
Create a structure or schema for the dataset to represent these elements and their relationships.
### Expected Behavior:

A clear methodology or set of steps for creating a heterogeneous graph dataset for railway scenarios.
Example code snippets or tools that can assist in building and labeling the dataset.
### Actual Behavior:

Currently, I do not have a clear starting point or methodology for creating the dataset.

**Comment by akihironitta** [comment]
I'd suggest taking a look at any examples under `torch_geometric.datasets` https://github.com/pyg-team/pytorch_geometric/tree/2.5.3/torch_geometric/datasets that are similar to your case.
